### PR TITLE
Add Groovy documentation

### PR DESCRIPTION
**Pull request by scherepn**
If you’re adding a new scraper, please ensure that you have:

- [x] Tested the scraper on a local copy of DevDocs
- [x] Ensured that the docs are styled similarly to other docs on DevDocs

- [x] Added these files to the <code>public/icons/*your_scraper_name*/</code> directory:
  - [x] `16.png`: a 16×16 pixel icon for the doc
  - [x] `16@2x.png`: a 32×32 pixel icon for the doc
  - [x] `SOURCE`: A text file containing the URL to the page the image can be found on or the URL of the original image itself

### CODE REVIEW

File: lib/docs/filters/groovy/clean_html.rb [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        end
+
+        css('ul.blockList > table').each do |node|
+          node.parent.before(node).remove
+        end
+
+        css('h3', 'h4').each do |node|
+          node.name = node.name.sub(/\d/) { |i| i.to_i - 1 }
+        end
+
```
Comment (simon04) [inline lib/docs/filters/groovy/clean_html.rb:82]:
```suggestion

        css('pre').each do |node|
          node['data-language'] = 'groovy'
        end
```

**Review by simon04 — COMMENTED** [review]
Can we link to out Java docs, i.e., replace the link https://docs.oracle.com/javase/8/docs/api/java/io/File.html with https://devdocs.io/openjdk~8/java/io/file?

**Comment by scherepn** [comment]
There's a long history/discussion with internal linking between documentation sets: https://github.com/freeCodeCamp/devdocs/issues/234

If we do decide to do something, I think this would be a good test case.

**Comment by scherepn** [comment]
I just noticed every package link on the homepage is broken.

File: lib/docs/filters/groovy/clean_html.rb (hunk 1 above)
Comment (scherepn) [reply lib/docs/filters/groovy/clean_html.rb:88]: Added. For some reason I thought Prism didn't support Groovy.

**Review by simon04 — APPROVED** [review]
Awesome, thank you!

**Pull request merged**